content/blog_posts/articles/hmr-in-adonisjs.md

```diff
@@ -0,0 +1,106 @@
+---
+summary: Hot Module Replacement (HMR) is now available in AdonisJS.
+---
+
+We're thrilled to announce that HMR is now available in AdonisJS!
+
+In short: HMR now allows us to modify the code of our controllers and their dependencies without having to restart the entire dev server each time. This results in a much faster feedback loop and a significantly improved development experience.
+
+## Hot Hook
+
+We've developed [Hot-Hook](https://github.com/Julien-R44/hot-hook), a framework-agnostic library that enables HMR in Node.JS with ESM. The library works with any framework. It's also important to note that Hot-hook **does not perform any static analysis, heavy code transformations, AST parsing, or bundling**.
+
+The approach is very simple but works perfectly. 
+
+:::note
+Make sure to read the documentation of [Hot-Hook](https://github.com/Julien-R44/hot-hook) which contains a lot of information about how it works, configuration, and some key concepts of the library.
+:::
+
+### How does it work?
+
+We won't go deeply in the details of the implementation, but here's a summary of how it works: 
+
+- Hot Hook is a [hook](https://nodejs.org/api/module.html#customization-hooks) for Node.js. In short: a hook is a way to execute some code when you import a module in your own codebase.
+- When you start your server with `node ace serve --hmr`, Hot Hook is loaded and will build a dependency graph of your application.
+- When you modify a file, Hot Hook will detect the change and will increment a version number for the file.
+- When the updated file is imported again ( because of a request, for example ), Hot Hook will append the version number to the `import` statement. That means we gonna `import('#controllers/users_controller.ts?version=2')`, and by doing this, we bypass the cache of the module and get the latest version of the file.
+
+This is a quick summary of how Hot Hook works, there is a bit more to it, but this is the main idea.
```

Comment:
```suggestion

```

Reply:
same why delete this ?

Reply:
Because this sentence is not needed in the article and it does not follow the tone of the article.